plugins/faustwp/includes/rest/callbacks.php

```diff
@@ -302,3 +314,53 @@ function wpac_authorize_permission_callback( \WP_REST_Request $request ) {
 
 	return false;
 }
+
+/**
+ * Handles permission checks for the telemetry decision REST route.
+ *
+ * @param \WP_REST_Request $request REST request object.
+ * @return bool Whether the user has permission to make telemetry decisions.
+ */
+function rest_telemetry_decision_permission_callback( \WP_REST_Request $request ) {
+	return current_user_can( 'manage_options' );
+}
+
+/**
+ * Handles user decisions for telemetry opt-in.
+ *
+ * @param \WP_REST_Request $request REST request object.
+ * @return \WP_REST_Response|\WP_Error
+ */
+function handle_rest_telemetry_decision_callback( \WP_REST_Request $request ) {
+	$body     = json_decode( $request->get_body(), true );
+	$decision = $body['decision'] ?? 'remind';
+	if ( ! in_array( $decision, array( 'yes', 'no', 'remind' ), true ) ) {
+		$decision = 'remind';
+	}
+	switch ( $decision ) {
+		case 'yes':
+			faustwp_update_setting( 'telemetry_reminder', '0' );
+			faustwp_update_setting( 'enable_telemetry', '1' );
+			break;
+		case 'no':
+			faustwp_update_setting( 'telemetry_reminder', '0' );
+			faustwp_update_setting( 'enable_telemetry', '0' );
+			break;
+		case 'remind':
+		default:
+			$date = new \DateTime( '+90 days', new \DateTimeZone( 'UTC' ) );
+			faustwp_update_setting( 'enable_telemetry', '0' );
+			faustwp_update_setting( 'telemetry_reminder', $date->getTimeStamp() );
+			break;
+	}
+
+	$response = array(
+		'decision' => $decision,
+		'settings' => array(
+			'enabled'  => faustwp_get_setting( 'enable_telemetry' ),
+			'reminder' => faustwp_get_setting( 'telemetry_reminder' ),
+		),
+		'success'  => true,
```

Comment:
`faustwp_update_setting()` does not return a value, hence this is currently always returning `true`. We need to better than this. A couple ideas:
- Fetch the value from the database after saving and compare to user-provided decision to determine success.
- Update `faustwp_update_setting()` to return what `update_option()` returns. This would require additional logic, because `update_option()` returns `false` if the values are the same as existing values, and we don't want to perceive this as a "failure" to update.